### DISCUSSION TITLE
Serial number of the device

### DISCUSSION THREAD

**Discussion by YashwanthLokam** [Yubico/libfido2#754] [Q&A] [answered]
Hi, I would like to know if I can get the Serial number of any Yubikey using this library. I have explored other language's Sdk but I didn't find anything. Please let me know if anyone has an idea on this.

**Comment by LDVG** [comment, marked as answer]
Hi,

The CTAP specification has no notion of a device serial number, so libfido2 does not implement any such methods. Generally speaking, you should not need a serial number to interface with a FIDO authenticator. Since you have not explained _why_ you need to retrieve the serial number, suggesting a solution is difficult. That said, for a YubiKey-specific solution, you can have a look at how e.g. [yubikey-manager](https://github.com/Yubico/yubikey-manager/blob/349852a24500944a2b4e01dc1204c53d39b34b90/yubikit/management.py#L428) does it.
